### ISSUE TITLE
Exception: BAD RESPONSE STATUS FROM API (0)

### ISSUE THREAD

**Issue by iamerwin** [WP2Static/static-html-output-plugin#27]
I made a storage zone with a pull zone linked to it at the friends from BunnCDN. They have the same names. 

After hitting the "Start static site export" button, the whole thing starts right up to transferring files to BunnyCDN which I can see in the storage zone. After some time this Failed during "Deploying files to BunnyCDN" error comes and the whole thing stops. No clue why and how and what, the export log says:

```
BAD RESPONSE STATUS FROM API (0)
Deployment: error encountered
Exception: BAD RESPONSE STATUS FROM API (0) in /example/home/wp-content/plugins/static-html-output-plugin/plugin/WP2Static/SitePublisher.php:2
Stack trace:
#0 /example/user/wp-content/plugins/static-html-output-plugin/plugin/deployers/BunnyCDN.php(2): WP2Static_SitePublisher->checkForValidResponses(0, Array)
WP2Static/wp2static#1 /example/user/wp-content/plugins/static-html-output-plugin/plugin/deployers/BunnyCDN.php(2): WP2Static_BunnyCDN->createFileInBunnyCDN()
WP2Static/wp2static#2 /example/user/wp-content/plugins/static-html-output-plugin/plugin/deployers/BunnyCDN.php(2): WP2Static_BunnyCDN->upload_files()
WP2Static/wp2static#3 /example/user/wp-content/plugins/static-html-output-plugin/plugin/deployers/BunnyCDN.php(2): WP2Static_BunnyCDN->__construct()
WP2Static/wp2static#4 /example/user/wp-content/plugins/static-html-output-plugin/plugin/WP2Static/Dispatcher.php(2): require_once('/example/home/...')
WP2Static/wp2static#5 /example/user/wp-content/plugins/static-html-output-plugin/wp2static.php(17): require_once('/example/home/...')
WP2Static/wp2static#6 /example/user/wp-settings.php(342): include_once('/example/home/...')
WP2Static/wp2static#7 /example/user/wp-config.php(90): require_once('/example/home/...')
WP2Static/wp2static#8 /example/user/wp-load.php(37): require_once('/example/home/...')
WP2Static/wp2static#9 /example/user/wp-admin/admin-ajax.php(22): require_once('/example/home/...')
WP2Static/wp2static#10 {main}
```

Well, it says a lot more. But this is where the error is.

Halp?

**Comment by thegulshankumar** [comment]
This is not a plugin issue but more likely a Browser/client side problem.

**There is one situation when this issue can happen.**

* You are using Google Chrome and you had /wp-admin/ password saved in your Browser.
* Now Chrome will autofill bad credential and cause API fail to respond.

@leonstafford Guess, this accidental issue can be prevented by replacing `type=password` with CSS approach as [suggested](https://stackoverflow.com/a/17769519) here.

**Comment by leonstafford** [comment]
^ yes, could be that issue. Also, the Bunny Test button now no longer reports false-positives. Closing this issue for now, as likely environmental.

**Comment by debrucer** [comment]
I know this is closed, but I had exactly this error message. This was one of two threads I found, and the other returned a value other than (0). So, this was the one. I am using Version 6.6.7 of your plugin (which to the best of my knowledge is still free). My WordPress just updated, and I suspect a lot of other things, too. I had not created a new static site for about three months, but had to change a page. Next came four days of debugging with two downed sites.

WordPress is being used for both sites, one is the master site, the other is a sub-site of the master. Not sure if I'm explaining that either correctly or clearly, but, you know what a WP multi-site setup is about. In four days of debugging, failed attempts, each trying one new thing, in the end, the only changes I made to fix the problem in both sites follows:

On the master site, I entered a restriction in the crawl menu ("/wp-content/uploads/sites"). In my case there were an empty site "2" and my sub-site named "3".  After this change the process went from taking an hour to fail, to finishing in ten minutes.

On the sub-site, I entered two lines in the force include, as follows:

/wp-content/uploads/sites/3/2021
/wp-content/uploads/sites/3/2022

Prior to this, the crawler was only getting 2020 and ignoring 2021 and 2022. With this change the process finished. I'm not sure how long as I fired it off and went to bed.

I tried not to dismiss the statement that this was environmental, and I tried three different browsers and all, also in incognito mode. With a recent MS Edge bug, all of my stored passwords were erased and the erasure propagated to all my other computers :(, so, there was no password stored.

I followed quite a few other links, saw things in PHP that could be the problem, tried to correct those, and reversed. None of those suggestion fit like the contents of this thread do.

I have used this program for three or four years and never had any problems before that weren't quickly fixable by removing the generated file and re-running the program one time. I suspect there's no development on any free version. I'm not a commercial entity, so huge subscription models don't work for me.

Great product! I'm sure you do well.
